From e8343f09917920b1e9119d3ecddb646c4121b906 Mon Sep 17 00:00:00 2001
From: Peter Marheine <pmarheine@chromium.org>
Date: Wed, 24 Jan 2024 11:17:57 +1100
Subject: [PATCH] tests: use $CC, not hard-coded gcc

Cross-compiling may need to use a different compiler, so don't hard code
references to plain `gcc` and instead determine the compiler to use from
the environment.
---
 tests/common.mak                            |  2 +-
 tests/gendiffcov/insensitive/insensitive.sh |  5 +++--
 tests/gendiffcov/simple/script.sh           |  5 +++--
 tests/gendiffcov/synthesize/synthesize.sh   |  5 +++--
 tests/genhtml/lambda/lambda.sh              |  3 ++-
 tests/genhtml/relative/relative.sh          |  3 ++-
 tests/lcov/branch/branch.sh                 | 12 +++++++-----
 tests/lcov/exception/exception.sh           | 10 ++++++----
 tests/lcov/extract/extract.sh               | 12 +++++++-----
 tests/lcov/format/format.sh                 |  8 +++++---
 tests/lcov/merge/merge.sh                   |  8 +++++---
 11 files changed, 44 insertions(+), 29 deletions(-)

diff --git a/tests/common.mak b/tests/common.mak
index 4c41bce..676d7b2 100644
--- a/tests/common.mak
+++ b/tests/common.mak
@@ -67,7 +67,7 @@ LCOVRC       := $(TOPDIR)lcovrc
 
 # Specify size for artificial info files (small, medium, large)
 SIZE         := small
-CC           := gcc
+export CC    ?= gcc
 
 export LCOV_TOOL := $(EXEC_COVER) $(BINDIR)/lcov
 export GENHTML_TOOL := $(EXEC_COVER) $(BINDIR)/genhtml
diff --git a/tests/gendiffcov/insensitive/insensitive.sh b/tests/gendiffcov/insensitive/insensitive.sh
index 0287f01..354e3fc 100755
--- a/tests/gendiffcov/insensitive/insensitive.sh
+++ b/tests/gendiffcov/insensitive/insensitive.sh
@@ -6,7 +6,8 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
-CXX='g++'
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -144,7 +145,7 @@ echo `which gcov`
 echo `which lcov`
 
 # old gcc version generates inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
 fi
diff --git a/tests/gendiffcov/simple/script.sh b/tests/gendiffcov/simple/script.sh
index 5947223..0d17df3 100755
--- a/tests/gendiffcov/simple/script.sh
+++ b/tests/gendiffcov/simple/script.sh
@@ -6,7 +6,8 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
-CXX='g++'
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
 LOCAL_COVERAGE=1
 KEEP_GOING=0
 
@@ -159,7 +160,7 @@ ${CXX} --coverage test.cpp
 echo `which gcov`
 echo `which lcov`
 # old version of gcc has inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
 fi
diff --git a/tests/gendiffcov/synthesize/synthesize.sh b/tests/gendiffcov/synthesize/synthesize.sh
index 983e073..3e3fb64 100755
--- a/tests/gendiffcov/synthesize/synthesize.sh
+++ b/tests/gendiffcov/synthesize/synthesize.sh
@@ -6,7 +6,8 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
-CXX='g++'
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -134,7 +135,7 @@ echo `which gcov`
 echo `which lcov`
 
 # old gcc version generates inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
 fi
diff --git a/tests/genhtml/lambda/lambda.sh b/tests/genhtml/lambda/lambda.sh
index b8ccb00..1280ad4 100755
--- a/tests/genhtml/lambda/lambda.sh
+++ b/tests/genhtml/lambda/lambda.sh
@@ -8,6 +8,7 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
+CC="${CC:-gcc}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -91,7 +92,7 @@ PARENT=`(cd .. ; pwd)`
 
 LCOV_OPTS="--branch $PARALLEL $PROFILE"
 # gcc/4.8.5 (and possibly other old versions) generate inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
     # and filter exception branches to avoid spurious differences for old compiler
diff --git a/tests/genhtml/relative/relative.sh b/tests/genhtml/relative/relative.sh
index b58ef3b..7128823 100755
--- a/tests/genhtml/relative/relative.sh
+++ b/tests/genhtml/relative/relative.sh
@@ -8,6 +8,7 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
+CC="${CC:-gcc}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -90,7 +91,7 @@ PARENT=`(cd .. ; pwd)`
 
 LCOV_OPTS="--branch $PARALLEL $PROFILE"
 # gcc/4.8.5 (and possibly other old versions) generate inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
     # and filter exception branches to avoid spurious differences for old compiler
diff --git a/tests/lcov/branch/branch.sh b/tests/lcov/branch/branch.sh
index bd6246a..cf2ed04 100755
--- a/tests/lcov/branch/branch.sh
+++ b/tests/lcov/branch/branch.sh
@@ -6,6 +6,8 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -93,7 +95,7 @@ PARENT=`(cd .. ; pwd)`
 
 LCOV_OPTS="--branch $PARALLEL $PROFILE"
 # gcc/4.8.5 (and possibly other old versions) generate inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
 fi
@@ -115,12 +117,12 @@ if [[ 1 == $CLEAN_ONLY ]] ; then
     exit 0
 fi
 
-if ! type g++ >/dev/null 2>&1 ; then
-        echo "Missing tool: g++" >&2
+if ! type ${CXX} >/dev/null 2>&1 ; then
+        echo "Missing tool: ${CXX}" >&2
         exit 2
 fi
 
-g++ -std=c++1y --coverage branch.cpp -o no_macro
+${CXX} -std=c++1y --coverage branch.cpp -o no_macro
 if [ 0 != $? ] ; then
     echo "Error:  unexpected error from gcc -o no_macro"
     exit 1
@@ -150,7 +152,7 @@ fi
 
 rm -f *.gcda *.gcno
 
-g++ -std=c++1y --coverage branch.cpp -DMACRO -o macro
+${CXX} -std=c++1y --coverage branch.cpp -DMACRO -o macro
 if [ 0 != $? ] ; then
     echo "Error:  unexpected error from gcc -o macro"
     exit 1
diff --git a/tests/lcov/exception/exception.sh b/tests/lcov/exception/exception.sh
index 0ab16a7..586b985 100755
--- a/tests/lcov/exception/exception.sh
+++ b/tests/lcov/exception/exception.sh
@@ -6,6 +6,8 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -94,7 +96,7 @@ PARENT=`(cd .. ; pwd)`
 
 LCOV_OPTS="--branch-coverage $PARALLEL $PROFILE"
 
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 
 rm -rf *.gcda *.gcno a.out *.info* *.txt* *.json dumper* testRC *.gcov *.gcov.*
 
@@ -106,12 +108,12 @@ if [[ 1 == $CLEAN_ONLY ]] ; then
     exit 0
 fi
 
-if ! type g++ >/dev/null 2>&1 ; then
-        echo "Missing tool: g++" >&2
+if ! type ${CXX} >/dev/null 2>&1 ; then
+        echo "Missing tool: ${CXX}" >&2
         exit 2
 fi
 
-g++ -std=c++1y --coverage exception.cpp
+${CXX} -std=c++1y --coverage exception.cpp
 if [ 0 != $? ] ; then
     echo "Error:  unexpected error from gcc"
     exit 1
diff --git a/tests/lcov/extract/extract.sh b/tests/lcov/extract/extract.sh
index 0cdec93..aa1ba2a 100755
--- a/tests/lcov/extract/extract.sh
+++ b/tests/lcov/extract/extract.sh
@@ -6,6 +6,8 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -93,7 +95,7 @@ PARENT=`(cd .. ; pwd)`
 
 LCOV_OPTS="--rc lcov_branch_coverage=1 $PARALLEL $PROFILE"
 # gcc/4.8.5 (and possibly other old versions) generate inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
     # and filter exception branches to avoid spurious differences for old compiler
@@ -114,12 +116,12 @@ if [[ 1 == $CLEAN_ONLY ]] ; then
     exit 0
 fi
 
-if ! type g++ >/dev/null 2>&1 ; then
-        echo "Missing tool: g++" >&2
+if ! type ${CXX} >/dev/null 2>&1 ; then
+        echo "Missing tool: ${CXX}" >&2
         exit 2
 fi
 
-g++ -std=c++1y --coverage extract.cpp
+${CXX} -std=c++1y --coverage extract.cpp
 if [ 0 != $? ] ; then
     echo "Error:  unexpected error from gcc"
     exit 1
@@ -399,7 +401,7 @@ let STRIP=$DEPTH+2
 mkdir -p separate/build
 mkdir -p separate/run
 mkdir -p separate/copy
-( cd separate/build ; g++ -std=c++1y --coverage ../../extract.cpp )
+( cd separate/build ; ${CXX} -std=c++1y --coverage ../../extract.cpp )
 cp separate/build/*.gcno separate/copy
 # make unwritable - so we don't allow lcov to write temporaries
 #  this emulates what happens when the build job is owned by one user,
diff --git a/tests/lcov/format/format.sh b/tests/lcov/format/format.sh
index a9f2b3f..0c70b45 100755
--- a/tests/lcov/format/format.sh
+++ b/tests/lcov/format/format.sh
@@ -8,6 +8,8 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -92,7 +94,7 @@ PARENT=`(cd .. ; pwd)`
 
 LCOV_OPTS="--branch $PARALLEL $PROFILE"
 # gcc/4.8.5 (and possibly other old versions) generate inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
     # and filter exception branches to avoid spurious differences for old compiler
@@ -109,8 +111,8 @@ if [[ 1 == $CLEAN_ONLY ]] ; then
     exit 0
 fi
 
-if ! type g++ >/dev/null 2>&1 ; then
-        echo "Missing tool: g++" >&2
+if ! type ${CXX} >/dev/null 2>&1 ; then
+        echo "Missing tool: ${CXX}" >&2
         exit 2
 fi
 
diff --git a/tests/lcov/merge/merge.sh b/tests/lcov/merge/merge.sh
index 3e5817c..65605a6 100755
--- a/tests/lcov/merge/merge.sh
+++ b/tests/lcov/merge/merge.sh
@@ -8,6 +8,8 @@ COVER=
 
 PARALLEL='--parallel 0'
 PROFILE="--profile"
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
 COVER_DB='cover_db'
 LOCAL_COVERAGE=1
 KEEP_GOING=0
@@ -85,7 +87,7 @@ PARENT=`(cd .. ; pwd)`
 
 LCOV_OPTS="--branch $PARALLEL $PROFILE"
 # gcc/4.8.5 (and possibly other old versions) generate inconsistent line/function data
-IFS='.' read -r -a VER <<< `gcc -dumpversion`
+IFS='.' read -r -a VER <<< `${CC} -dumpversion`
 if [ "${VER[0]}" -lt 5 ] ; then
     IGNORE="--ignore inconsistent"
     # and filter exception branches to avoid spurious differences for old compiler
@@ -102,8 +104,8 @@ if [[ 1 == $CLEAN_ONLY ]] ; then
     exit 0
 fi
 
-if ! type g++ >/dev/null 2>&1 ; then
-        echo "Missing tool: g++" >&2
+if ! type ${CXX} >/dev/null 2>&1 ; then
+        echo "Missing tool: ${CXX}" >&2
         exit 2
 fi
 
-- 
2.43.0.429.g432eaa2c6b-goog

